src/tiledb/cloud/vcf/ingestion.py

```diff
@@ -255,6 +255,44 @@ def read_uris_udf(
         return result
 
 
+def read_metadata_uris_udf(
+    dataset_uri: str,
+    *,
+    config: Optional[Mapping[str, Any]] = None,
+    metadata_uri: str,
+    metadata_attr: str = "uri",
+    max_files: Optional[int] = None,
+    verbose: bool = False,
+) -> Sequence[str]:
+    """
+    Read a list of URIs from a TileDB array. The URIs will be read from the
+    attribute specified in the `metadata_attr` argument.
+
+    :param dataset_uri: dataset URI
+    :param config: TileDB config, defaults to None
+    :param metadata_uri: metadata array URI
+    :param metadata_attr: name of metadata attribute containing URIs, defaults to "uri"
+    :param max_files: maximum number of URIs returned, defaults to None
+    :param verbose: verbose logging, defaults to False
+    :return: list of URIs
+    """
+    logger = get_logger_wrapper(verbose)
+
+    with tiledb.scope_ctx(config):
+        with Profiler(group_uri=dataset_uri, group_member=LOG_ARRAY) as prof:
+            with tiledb.open(metadata_uri) as A:
+                df = A.query(dims=[], attrs=[metadata_attr]).df[:]
+            results = df[metadata_attr].to_list()
```

Comment:
What about just doing `A.query(dims=[], atts=[metadata_attr])[:]` then you can avoid the `.to_list()` from pandas.

Reply:
It's a little more complicated. I pushed a change that avoids using pandas.